src/transformers/cache_utils.py

```diff
@@ -299,6 +299,22 @@ class DynamicCache(Cache):
 
     It stores the Key and Value states as a list of tensors, one for each layer. The expected shape for each tensor is
     `[batch_size, num_heads, seq_len, head_dim]`.
+
+    Example:
+
+        ```python
+        >>> from transformers import AutoTokenizer, AutoModelForCausalLM, DynamicCache
+
+        >>> model = AutoModelForCausalLM.from_pretrained("openai-community/gpt2")
+        >>> tokenizer = AutoTokenizer.from_pretrained("openai-community/gpt2")
+
+        >>> inputs = tokenizer(text="My name is GPT2", return_tensors="pt")
+
+        >>> # Prepare a cache class and pass it to model's forward
+        >>> past_key_values = DynamicCache()
+        >>> outputs = model(**inputs, past_key_values=past_key_values, use_cache=True)
+        >>> past_kv_length = outputs.past_key_values # access cache filled with key/values from generation
```

Comment:
Missing the output in these examples, so we can doctest :P

Reply:
Not sure what should be the output here, as we are not printing anything. I wasn't very pro of getting kv values or logits slice, as it's prone to numerical precision errors

Reply:
Yeah, I was afraid to cause more numerical precision fails, but can be added for the sake of testing

Reply:
Yeah, I was afraid to cause more numerical precision fails, but can be added for the sake of testing